### ISSUE TITLE
updateTriggers doesn't work when transitioning between 0 and null

### ISSUE THREAD

**Issue by supersonicclay** [uber/deck.gl#2068]
#### Actual Result
Data accessor isn't called when `updateTriggers` value transitions from `0` to `null` (or vise versa).

#### Expected Result
Data accessor should be called by doing a strict equality check (`===`).

#### Reproduce Steps
```
getColor={() =>
  this.state.colorIndex === null ? [255, 255, 255] : COLORS[this.state.colorIndex]
}
updateTriggers={{
  getColor: this.state.colorIndex
}}
```
When `colorIndex` transitions from 1 to 2, `getColor` is called. When `colorIndex` transitions from `0` to `null` (or vise versa), `getColor` is not called.

#### Code Example
In branch `update-trigger-bug`

#### To Do List 
  - [ ] Add label and assign to milestone
  - [ ] Coding
  - [ ] Test

**Comment by supersonicclay** [comment]
Likely somewhere in https://github.com/uber/deck.gl/blob/master/modules/core/src/lifecycle/props.js#L39

**Comment by supersonicclay** [comment]
Workaround is to use an array instead:

```
updateTriggers={{
  getColor: [this.state.colorIndex]
}}
```

**Comment by ibgreen** [comment]
Let's fix this even if there is a workaround.

**Comment by tgorkin** [comment]
Fixed in #2475
